editor/src/main/java/io/github/rosemoe/sora/widget/CodeEditor.java

```diff
@@ -613,6 +727,223 @@ private void initialize(AttributeSet attrs, int defStyleAttr, int defStyleRes) {
+            }
+            setColorScheme(editorColorScheme);
+        }
+        if (generalItem.cursorBlinkPeriod != 0) setCursorBlinkPeriod(generalItem.cursorBlinkPeriod);
+        if (generalItem.cursorWidth != 0) setCursorWidth(generalItem.cursorWidth);
+
+        if (generalItem.dividerWidth != 0) setDividerWidth(generalItem.dividerWidth);
+        if (generalItem.dividerMarginLeft != 0 || generalItem.dividerMarginRight != 0) setDividerMargin(generalItem.dividerMarginLeft, generalItem.dividerMarginRight);
+
+        if (generalItem.edgeEffectColor != 0) setEdgeEffectColor(generalItem.edgeEffectColor);
+
+        if (generalItem.lineNumberAlign != 0) {
+            Paint.Align align = null;
+            switch (generalItem.lineNumberAlign) {
+                case GeneralItem.ALIGN_LEFT:
+                    align = Paint.Align.LEFT;
+                    break;
+                case GeneralItem.ALIGN_RIGHT:
+                    align = Paint.Align.RIGHT;
+                    break;
+                case GeneralItem.ALIGN_CENTER:
+                    align = Paint.Align.CENTER;
+                    break;
+            }
+            setLineNumberAlign(align);
+        }
+        if (generalItem.lineNumberMarginLeft != 0) setLineNumberMarginLeft(generalItem.lineNumberMarginLeft);
+        if (generalItem.lineSpacingExtra != 0) setLineSpacingExtra(generalItem.lineSpacingExtra);
+        if (generalItem.linePanelInfoTextSize != 0) setLineInfoTextSize(generalItem.linePanelInfoTextSize);
+        if (generalItem.linePanelPosition != 0) {
+            int pos = 0;
+            switch (generalItem.linePanelPosition) {
+                case GeneralItem.LN_PANEL_POSITION_LEFT:
+                    pos = LineInfoPanelPosition.LEFT;
+                    break;
+                case GeneralItem.LN_PANEL_POSITION_TOP:
+                    pos = LineInfoPanelPosition.TOP;
+                    break;
+                case GeneralItem.LN_PANEL_POSITION_RIGHT:
+                    pos = LineInfoPanelPosition.RIGHT;
+                    break;
+                case GeneralItem.LN_PANEL_POSITION_BOTTOM:
+                    pos = LineInfoPanelPosition.BOTTOM;
+                    break;
+                case GeneralItem.LN_PANEL_POSITION_CENTER:
+                    pos = LineInfoPanelPosition.CENTER;
+                    break;
+            }
+            setLnPanelPosition(pos);
+        }
+        if (generalItem.linePanelPositionMode != 0) {
+            int mode = 0;
+            switch (generalItem.linePanelPositionMode) {
+                case GeneralItem.LN_PANEL_POSITION_MODE_FIXED:
+                    mode = LineInfoPanelPositionMode.FIXED;
+                    break;
+                case GeneralItem.LN_PANEL_POSITION_MODE_FOLLOW:
+                    mode = LineInfoPanelPositionMode.FOLLOW;
+                    break;
+            }
+            setLnPanelPositionMode(mode);
+        }
+
+        if (generalItem.nonPrintablePaintingFlags != 0) {
```

Comment:
Value of `generalItem.nonPrintablePaintingFlags` is not obtained from attributes.